pii-detector-service/tests/unit/test_sonar_refactoring_safety_net.py

```
@@ -0,0 +1,980 @@
+    ):
+        detector = self._create_detector(mock_toml_config_with_provider)
+        detector._analyzer = None
+
+        with patch(
+            "pii_detector.infrastructure.detector.presidio_detector.NlpEngineProvider",
+            side_effect=RuntimeError("NLP provider failed"),
+        ), patch(
+            "pii_detector.infrastructure.detector.presidio_detector.AnalyzerEngine",
+            side_effect=RuntimeError("All engines failed"),
+        ):
+            with pytest.raises(RuntimeError):
```

Ruff (0.15.4)

[error] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)